plugins/cody-chat/src/com/sourcegraph/cody/CodyResources.java

```diff
@@ -1,32 +1,60 @@
 package com.sourcegraph.cody;
 
 import java.io.IOException;
+import java.io.InputStream;
 import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import org.eclipse.core.runtime.Platform;
 
 public class CodyResources {
 
-  public static String loadIndex() {
-    return loadResourceString("/resources/index.html");
-  }
+  private static final ResourcePath WEBVIEW_ASSETS = ResourcePath.of("/resources/webviews");
+  private static final ResourcePath AGENT_ASSETS = ResourcePath.of("/resources/cody-agent");
+  private static final ResourcePath NODE_BINARIES_PATH =
+      ResourcePath.of("/resources/node-binaries");
 
-  public static String loadCodyIndex() {
-    String content = loadResourceString("/resources/cody-webviews/index.html");
-    return content
-        .replace("'self'", "'self' https://*.sourcegraphstatic.com")
+  private static byte[] loadWebviewIndexBytes() {
+    String html = loadResourceString(WEBVIEW_ASSETS.resolve("index.html").toString());
+    var start =
+        indexOrOrCrash(
```

Comment:
not sure if this would help, but agent will send a notification event with the html you are getting from the local file here: https://sourcegraph.com/github.com/sourcegraph/cody@main/-/blob/vscode/src/jsonrpc/agent-protocol.ts?L485